src/main.cpp

```diff
@@ -48,6 +49,69 @@ namespace po = boost::program_options;
 using namespace aalwines;
 using namespace pdaaal;
 
+template<typename W_FN>
+bool do_verification(stopwatch& compilation_time, stopwatch& reduction_time, stopwatch& verification_time,
+        Query& q, Query::mode_t m, Network& network, bool no_ip_swap, std::pair<size_t,size_t>& reduction, size_t tos,
+        bool need_trace, size_t engine, Moped& moped, SolverAdapter& solver, utils::outcome_t& result,
+        std::vector<pdaaal::TypedPDA<Query::label_t>::tracestate_t >& trace, std::stringstream& proof,
+        const W_FN& weight_fn) {
+    compilation_time.start();
+    q.set_approximation(m);
+    NetworkPDAFactory factory(q, network, no_ip_swap, weight_fn);
+    auto pda = factory.compile();
+    compilation_time.stop();
+
+    reduction_time.start();
+    reduction = Reducer::reduce(pda, tos, pda.initial(), pda.terminal());
+    reduction_time.stop();
+
+    verification_time.start();
+    bool engine_outcome;
+    switch(engine) {
+        case 1:
+            engine_outcome = moped.verify(pda, need_trace);
```

Comment:
Maybe we want an error here if a weight-function is given, but not respected (same for engine 3)

Reply:
Line 313 of main.cpp reports such an error. Is that enough?